### PR TITLE
basic LAPI TT tests for prefetchContractKeys 

### PR DESCRIPTION
**Pull request by andreaslochbihler-da**

### CODE REVIEW

File: sdk/ledger-test-tool/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_17/PrefetchContractKeysIT.scala [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  import PrefetchContractKeysIT._
+
+  test(
+    "CSprefetchContractKeysBasic",
+    "Explicit contract key prefetches are accepted",
+    allocate(SingleParty),
+  )(implicit ec => { case Participants(Participant(ledger, party)) =>
+    val prefetch = WithKey.byKey(party).toPrefetchKey().toProtoInner
+    val request = ledger
+      .submitAndWaitRequest(party, new WithKey(party).create.commands)
```
Comment (mziolekda) [inline sdk/ledger-test-tool/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_17/PrefetchContractKeysIT.scala:42]: I think we could create an invalid key if we went a la nature then we would be able to see the transaction getting rejected WDYT?

Comment (mziolekda) [reply sdk/ledger-test-tool/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_17/PrefetchContractKeysIT.scala:44]: Added it in 93ff12b7a6249d2ec9a01696ac8994c436dfba2b

File: sdk/ledger-test-tool/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_17/PrefetchContractKeysIT.scala
Comment (tudor-da) [inline sdk/ledger-test-tool/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_17/PrefetchContractKeysIT.scala:1]: What happens if keys are prefetched superfluously (not eventually needed in interpretation)? I'd be explicit in asserting this spec. Not a blocker for this PR though

**Review by tudor-da — APPROVED** [review]
lgtm!

**Pull request merged**